layer/components/Amount/Base.vue

```diff
@@ -0,0 +1,132 @@
+    roundingMode?: BigNumber.RoundingMode
+    amount: string | number | BigNumberInBase
+  }>(),
+  {
+    showSmallerThan: true,
+    decimals: DEFAULT_DISPLAY_DECIMALS,
+    roundingMode: BigNumber.ROUND_DOWN,
+    subscriptDecimals: DEFAULT_SUBSCRIPT_DECIMALS,
+    abbreviationThreshold: DEFAULT_ABBREVIATION_THRESHOLD,
+    subscriptThresholdDecimals: DEFAULT_SUBSCRIPT_THRESHOLD_DECIMALS
+  }
+)
+
+const amountToBigNumber = computed(() => new BigNumberInBase(props.amount || 0))
+const isNegative = computed(() => amountToBigNumber.value.lt(0))
+const absoluteAmount = computed(() => amountToBigNumber.value.abs())
+
+const minDecimalThreshold = computed(() =>
+  new BigNumberInBase(1).div(Math.pow(10, props.decimals))
+)
+
+const shouldHaveSmallerThan = computed(() => {
+  const amount = absoluteAmount.value
+  const nIsLowerThanDecimalThreshold = amount.lt(minDecimalThreshold.value)
+
+  return nIsLowerThanDecimalThreshold && amount.gt(0)
+})
+
+const abbreviatedAmount = computed(() => {
+  if (!props.shouldAbbreviate) {
+    return false
+  }
+
+  const amount = absoluteAmount.value
+
+  const nIsBiggerThanThreshold = amount.gte(props.abbreviationThreshold)
+
+  if (!!props.abbreviationThreshold && nIsBiggerThanThreshold) {
+    return abbreviateNumber(amount.toNumber())
+  }
+
+  return false
+})
+
+const subscriptedAmount = computed(() => {
+  const [integerPart, decimalPart] = absoluteAmount.value.toFixed().split('.')
+
+  if (!decimalPart || !props.useSubscript || absoluteAmount.value.gte(1)) {
+    return false
+  }
+
+  let nOfZeros = 0
+
+  for (let i = 0; i < decimalPart.length; i++) {
+    if (decimalPart[i] === '0') {
+      nOfZeros++
+    } else {
+      break
+    }
+  }
+
+  if (
+    nOfZeros > props.subscriptThresholdDecimals ||
+    nOfZeros > props.decimals
+  ) {
+    const subscriptAmount = new BigNumberInBase(decimalPart.replace(/^0+/, ''))
+      .toFixed(0)
+      .slice(0, props.subscriptDecimals)
+
+    const integerAmount = new BigNumberInBase(integerPart).toFormat(0)
+
+    return integerAmount + '.0<sub>' + nOfZeros + '</sub>' + subscriptAmount
+  }
+
+  return false
+})
+
+const formattedAmount = computed(() => {
+  const amount = absoluteAmount.value
+
+  if (props.noTrailingZeros && props.decimals > 0) {
+    const result = amount
+      .toFormat(props.decimals, props.roundingMode)
+      .replace(/(\.\d*?[1-9])0+$/g, '$1')
+      .replace(/\.0+$/, '')
+
+    return result || '0'
+  }
+
+  const formattedAmount = amount.toFormat(props.decimals, props.roundingMode)
+
+  return formattedAmount
+})
+</script>
+
+<template>
+  <span>
+    <span v-if="isNegative">-</span>
+    <slot name="prefix" />
+
+    <span v-if="showZeroAsEmDash && amountToBigNumber.eq(0)"> &mdash; </span>
+    <span v-else-if="showSmallerThan && shouldHaveSmallerThan"
+      >&lt;{{ minDecimalThreshold.toFormat() }}
+    </span>
+    <span v-else-if="subscriptedAmount" v-html="subscriptedAmount"></span>
```

Comment:
think we don't need `showSmallerThan`, if we have `useSubscript` props as `true`, then we won't have `&lt;`
but if normal number, we'll kind of always have `&lt;` right

Reply:
oh, so is it `useSubscript` alone won't help (i was imagining `useSubscript true` when don't have space, then for asset amount `useSubscript false`)